doc/extensions/zephyr/link-roles.py

```diff
@@ -22,8 +26,17 @@ def get_github_rev():
 def setup(app):
     rev = get_github_rev()
 
-    # links to files or folders on the GitHub
-    baseurl = 'https://github.com/zephyrproject-rtos/zephyr'
+    # try to get url from West; this adds compatibility with repos
+    # located elsewhere
+    if WestManifest is not None:
+        baseurl = WestManifest.from_file().get_projects(['zephyr'])[0].url
+    else:
+        baseurl = None
+
+    # or fallback to default
+    if baseurl is None:
```

Comment:
That would fallback when `WestManifest` was not imported. But the reason I put those lines there is that `baseurl = WestManifest.from_file().get_projects(['zephyr'])[0].url` is `None` for my Zephyr tree, but it has the proper url for the `sdk-zephyr` tree. When I looked into it, for the Zephyr tree there is a `ManifestProject` object with no URL, while every other project is a `Project` object. For the `sdk-zephyr` "zephyr" is also a `Project` afair with a proper url. Didn't know if this behavior was on purpose or not...

Reply:
Ah... yeah, OK, let's just leave this as is. The situation is a bit of a mess that needs cleaning, see https://github.com/zephyrproject-rtos/west/issues/327 if you care about the details.

Reply:
Just to add to the previous comment, on NCS `WestManifest.from_file().get_projects(['nrf'])[0].url` is `None` because in that project `nrf` is a `ManifestProject` and is also not getting a `url`.